cmd/ftl/cmd_serve.go

```diff
@@ -116,3 +189,33 @@ func incrementPort(baseURL *url.URL) (*url.URL, error) {
 	newURL.Host = fmt.Sprintf("%s:%d", baseURL.Hostname(), newPort+1)
 	return &newURL, nil
 }
+
+func pollContainerHealth(ctx context.Context, containerName string, timeout time.Duration) error {
+	logger := log.FromContext(ctx)
+	logger.Infof("Waiting for %s to be healthy", containerName)
+
+	pollCtx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	for {
+		select {
+		case <-pollCtx.Done():
+			return errors.New("timed out waiting for container to be healthy")
+		default:
```

Comment:
Use `case <-time.After(100 * time.Millisecond):` instead of this and the sleep below.